src/image.jl

```diff
@@ -179,6 +179,17 @@ function write(f::FITS, data::Array{T};
     nothing
 end
 
+"""
+    write(hdu::ImageHDU, data::Array)
+
+Write data to an existing image HDU.
+"""
+function write(hdu::ImageHDU, data::Array{T}) where T
+    fits_assert_open(hdu.fitsfile)
```

Comment:
One thing missing here: moving to the right HDU: `fits_movabs_hdu(hdu.fitsfile, hdu.ext)`